docs/siem/detections/api/lists/api-create-list-item.asciidoc

```diff
@@ -0,0 +1,82 @@
+[[lists-api-create-list-item]]
+=== Create list item
+
+Creates a list item and associates it with the specified
+<<lists-api-create-container, list container>>.
+
+All list items in the same list container must be the same type. For example,
+each list item in an `ip` list container must exclude a specific IP address.
+
+NOTE: Before creating list items, you must create a list container.
+
+==== Request URL
+
+`POST <kibana host>:<port>/api/lists/items`
+
+==== Request body
+
+A JSON object with these fields:
+
+[width="100%",options="header"]
+|==============================================
+|Name |Type |Description |Required
+
+|`id` |String |Unique identifier of the list item. |No, automatically created
+when it is not provided.
+|`list_id` |String |ID of the associated <<lists-api-create-container, list container>>. |Yes
+|`meta` |Object |Placeholder for metadata about the list item. |No
+|`value` |String |The value to be excluded. |Yes
+
+|==============================================
+
+===== Example requests
+
+Adds an IP address to the `internal-ip-excludes` list container:
+
+[source,console]
+--------------------------------------------------
+POST api/lists/items
+{
+  "id": "internal-ip-address",
+  "list_id": "internal-ip-excludes",
+  "value": "10.0.0.1"
+}
```

Comment:
What would be important would be to show two examples for the `Add` and `Get` around ip and ip-range because the security user is going to ask about both of these concepts.

The differences and why they're important is that if you upload and use a list that consists of only ip then you can actually do a `GET` utilizing a CIDR.

For example using some of the scripts for illustration and ease:

```ts
❯ ./import_list_items_by_filename.sh ip ./lists/files/ips.txt
{
  "_version": "Wzg2LDFd",
  "id": "ips.txt",
  "created_at": "2020-08-04T20:35:54.416Z",
  "created_by": "yo",
  "description": "File uploaded from file system of ips.txt",
  "immutable": false,
  "name": "ips.txt",
  "tie_breaker_id": "a9c3a101-d121-45a6-815a-07130e115f47",
  "type": "ip",
  "updated_at": "2020-08-04T20:35:54.416Z",
  "updated_by": "yo",
  "version": 1
}
```

We can see the contents of the list like so:
```ts
❯ ./export_list_items.sh ips.txt
127.0.0.4
127.0.0.9
127.0.0.5
127.0.0.1
127.0.0.3
127.0.0.7
127.0.0.8
127.0.0.6
127.0.0.2
```

And now I can use a CIDR to query the list and get back _an array_ of values up to a maximum of 10k items. If there are more than 10k items on a CIDR type query then it will cap out at 10k (which is good for us to document). I had to put the 10k cap out with a TODO block that later we have to re-stream more than that :-/ hehe. But you know, for completeness they should get most of what they want from the feature.

Here is a 32 CIDR that is returning only one element which is 127.0.0.1

```ts
❯ ./get_list_item_by_value.sh ips.txt 127.0.0.1/32
[
  {
    "created_at": "2020-08-04T20:35:55.194Z",
    "created_by": "yo",
    "id": "mbwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "47d3d57b-c5fe-4948-8d73-d147cfcf5c22",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.194Z",
    "updated_by": "yo",
    "value": "127.0.0.1"
  }
]
```

Here is a 30 CIDR which is going to return three things:

```ts
❯ ./get_list_item_by_value.sh ips.txt 127.0.0.1/30
[
  {
    "created_at": "2020-08-04T20:35:55.194Z",
    "created_by": "yo",
    "id": "mbwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "47d3d57b-c5fe-4948-8d73-d147cfcf5c22",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.194Z",
    "updated_by": "yo",
    "value": "127.0.0.1"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "mrwwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "e77c0764-98b8-4a6a-8a04-bb0bf62ff56c",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.2"
  },
  {
    "created_at": "2020-08-04T20:35:55.195Z",
    "created_by": "yo",
    "id": "m7wwu3MBM_3LKPKZDKzB",
    "list_id": "ips.txt",
    "tie_breaker_id": "4838d3d7-9958-4f97-8833-63fa07aa5a0b",
    "type": "ip",
    "updated_at": "2020-08-04T20:35:55.195Z",
    "updated_by": "yo",
    "value": "127.0.0.3"
  }
]
```

Why is this part important in relation to the exception list feature? It's important because they can upload a list of plain ip's and then if they have an ECS field or string field which contains a CIDR and are looking within a list it should be able to exclude anything using that CIDR when it finds it inside the list.

ECS does not seem to indicate a CIDR type field yet ... I will admit that:
https://github.com/elastic/ecs/issues/86

But I still think this is important in case they are storing somewhere CIDR strings and using those to go against a list (which they might be in some companies and places)

Reply:
Ok second example which I flip this around and show an `ip_range` and them utilizing it would be like so:

```ts
❯ ./import_list_items_by_filename.sh ip_range ./lists/files/ip_range.txt
{
  "_version": "Wzg3LDFd",
  "id": "ip_range.txt",
  "created_at": "2020-08-04T20:36:26.406Z",
  "created_by": "yo",
  "description": "File uploaded from file system of ip_range.txt",
  "immutable": false,
  "name": "ip_range.txt",
  "tie_breaker_id": "9de426bf-fdad-461a-881e-c4b09485df71",
  "type": "ip_range",
  "updated_at": "2020-08-04T20:36:26.406Z",
  "updated_by": "yo",
  "version": 1
}
```

Then we have these values if we do an export:
```ts
❯ ./export_list_items.sh ip_range.txt
192.168.0.5-192.168.0.8
192.168.0.10-192.168.1.20
192.168.1.5-192.168.1.8
192.168.1.0-192.168.1.3
92.168.0.10-192.168.0.20
192.168.0.1-192.168.0.3
```

Now, what we can do here is the opposite of above, we can query and use plain IP's in fields such as `source.ip` or `destination.ip` with the exceptions feature or just plain query it like so:

```ts
❯ ./get_list_item_by_value.sh ip_range.txt 192.168.0.11
[
  {
    "created_at": "2020-08-04T20:36:27.284Z",
    "created_by": "yo",
    "id": "TwEwu3MB7Zv7jX8-iuYZ",
    "list_id": "ip_range.txt",
    "tie_breaker_id": "ad94aba5-4ecc-4038-b967-96f0c84ff170",
    "type": "ip_range",
    "updated_at": "2020-08-04T20:36:27.284Z",
    "updated_by": "yo",
    "value": "92.168.0.10-192.168.0.20"
  },
  {
    "created_at": "2020-08-04T20:36:27.284Z",
    "created_by": "yo",
    "id": "UgEwu3MB7Zv7jX8-iuYZ",
    "list_id": "ip_range.txt",
    "tie_breaker_id": "19b06de1-ce1d-4793-bce3-395c74861683",
    "type": "ip_range",
    "updated_at": "2020-08-04T20:36:27.284Z",
    "updated_by": "yo",
    "value": "192.168.0.10-192.168.1.20"
  }
]
```

And that IP can show up in two entries/ranges which means we have found it in two different spots and there is some overlap. This has the same 10k limit as the one above it.

These two combined make it easier for users to reserve different blocks of either IPv4 or IPv6 and use CIDR or ranges depending on what their goals are to be able to do exclusions based on their network topology.